### PR TITLE
feat: Brush up repo for publishing

### PR DESCRIPTION
**Pull request by slowli**
# What ❔

Starts brushing up the repo for publishing:

- Better structures the repo directories.
- Renames crates in a uniform way + uses workspace dependencies.
- Structures imports using well-defined configuration.
- Reduces visibility of some items.
- Adds more crate metadata, e.g. licenses and readmes.

## Why ❔

Part of preparations for publishing the library crates.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and `cargo clippy`.

### CODE REVIEW

File: crates/vm2/src/heap.rs [hunk 1]
```diff
@@ -69,55 +69,8 @@ impl Heap {
...
-
-#[inline(always)]
-fn address_to_page_offset(address: u32) -> (usize, usize) {
-    let offset = address as usize;
-    (offset >> 12, offset & (HEAP_PAGE_SIZE - 1))
-}
-
-impl HeapInterface for Heap {
-    fn read_u256(&self, start_address: u32) -> U256 {
+    // TODO: reduce visibility once `multivm` uses `StateInterface` APIs
```
Comment (slowli) [inline crates/vm2/src/heap.rs:72]: I plan to do this in a follow-up PR to keep the scope of this one reasonable.

File: crates/vm2-interface/src/lib.rs [hunk 2]
```diff
@@ -23,9 +23,9 @@
 //! version that you publish and import it from the previous version instead.
 //!
 //! This is how you would add a new method to StateInterface and a new opcode.
+//!
 //! ```
-//! # use eravm_stable_interface as eravm_stable_interface_v1;
```
Comment (joonazan) [inline crates/vm2-interface/src/lib.rs:27]: I'd keep this renaming to make clearer that you are supposed to have the older version as a dependency.

File: crates/vm2/src/bitset.rs [hunk 3]
```diff
@@ -1,5 +1,6 @@
+/// Bitset with 1,024 elements.
```
Comment (joonazan) [inline crates/vm2/src/bitset.rs:1]: 2^16 actually. Note that the array contains u64.

File: crates/vm2/src/heap.rs (hunk 1 above)
Comment (joonazan) [reply crates/vm2/src/heap.rs:72]: In theory, a lot of fields in state could be made private using this method as well.

**Review by joonazan — CHANGES_REQUESTED** [review]
Fix the incorrect 1024 comment and I'll approve.

File: crates/vm2/src/heap.rs (hunk 1 above)
Comment (slowli) [reply crates/vm2/src/heap.rs:72]: Yep, that's the plan.

**Review by joonazan — APPROVED** [review]

**Pull request merged**